tsdb/engine/tsm1/file_store.go

```diff
@@ -123,12 +124,16 @@ func (f FileStat) ContainsKey(key string) bool {
 	return f.MinKey >= key || key <= f.MaxKey
 }
 
-func NewFileStore(dir string) *FileStore {
+func NewFileStore(config tsdb.ShardConfig) *FileStore {
```

Comment:
Same criticism from above applies here too. It seems `NewEngine` is the only one that should be modified.